user.go

```diff
@@ -1040,8 +1040,9 @@ func (user *User) e2eeEventHandler(rawEvt any) {
 		user.BridgeState.Send(user.waState)
 		go user.sendMarkdownBridgeAlert(context.TODO(), "Error in WhatsApp connection: %s", evt.PermanentDisconnectDescription())
 	case events.PermanentDisconnect:
-		cf, ok := evt.(*events.ConnectFailure)
-		if ok && cf.Reason == events.ConnectFailureLoggedOut && user.canReconnect() {
+		cf, ok := evt.(*events.LoggedOut)
+		if ok && cf.Reason == events.ConnectFailureLoggedOut && !cf.OnConnect && user.canReconnect() {
```

Comment:
The !cf.OnConnect avoids reconnecting if the 401 comes from the user logging out from facebook totally (not just the e2e/WA socket log out described in the PR).

In that scenario `messagix.Event_PermanentError` is also triggered.